### DISCUSSION TITLE
Base64 Image Support

### DISCUSSION THREAD

**Discussion by ricardov03** [spatie/ray#498] [Ideas]
Could be nice to validate and preview Base64 images from ray.
**Example:**
ray()->image($string)->base64()

**Comment by patinthehat** [comment]
@ricardov03 When you say "base64 images", do you mean just a plain b64-encoded string, or a string like `data:image/png;base64, ...`?

**Reply by ricardov03** [reply]
[Base64](https://en.wikipedia.org/wiki/Base64) the group of binary-to-text encoding schemes that represent binary data.

**Reply by ricardov03** [reply]
I think I don't get the question before. I recommend using the most commonly used encoded types: Plain Text and Data URI.

**Reply by patinthehat** [reply]
I'm happy to build out the functionality and submit a PR, but I can't promise it'll be accepted. :smile:

**Reply by ricardov03** [reply]
Thanks so much for the interest. I hope they proceed with the PR. Do you find this functionality useful too?

**Reply by patinthehat** [reply]
I can see how it could be useful. In projects that store images as base64-encoded values, for instance. :wink:

**Reply by patinthehat** [reply]
@ricardov03 I've submitted a PR for this feature: https://github.com/spatie/ray/pull/499

**Reply by ricardov03** [reply]
Thanks, @patinthehat! Best regards. I'm going to review this to learn how to do it in the future.
